processing/src/main/java/io/druid/segment/filter/Filters.java

```diff
@@ -157,6 +165,45 @@ public void remove()
     );
   }
 
+  public static ValueMatcher getLongValueMatcher(
+      final LongColumnSelector longSelector,
+      Comparable value
+  )
+  {
+    if (value == null) {
+      return new BooleanValueMatcher(false);
+    }
+
+    final Long longValue = Longs.tryParse(value.toString());
+    if (longValue == null) {
+      return new BooleanValueMatcher(false);
+    }
+
+    return new ValueMatcher()
+    {
+      @Override
+      public boolean matches()
+      {
+        return longSelector.get() == longValue;
```

Comment:
Again, I'm not sure if the JVM is smart enough to avoid boxing overhead on each comparison. If not, you should unbox it manually before creating the ValueMatcher.